metrics_utility/management/commands/build_report.py

```diff
@@ -61,43 +85,34 @@ def init_logging(self):
         handler.setLevel(logging.DEBUG)
         handler.setFormatter(logging.Formatter('%(message)s'))
         self.logger.addHandler(handler)
-        self.logger.propagate = False
-
-    def handle(self, *args, **options):
+        self.logger.propagate = True
+
+    def _parse_param(
+        self,
+        param_name,
+        options,
+    ):
+        param = options.get(param_name)
+        if (options.get('month') is not None or param is None) and param_name != 'ephemeral':
+            return None
+        if param:
+            return parse_date_param(param) if re.match(DATE_REGEX_PATTERN, param) else parse_number_of_days(param)
```

Comment:
I think we may need to simplify this function, it may be doing too much...

Right now, it will try to parse `ephemeral` as a date if it looks like a date, which shouldn't be valid.

Also, `parse_date_param` returns a datetime, while `parse_number_of_days` returns an integer .. and `opt_ephemeral` expects a string, because it still calls `parse_number_of_days` later .